server/src/test/java/org/opensearch/search/SearchServiceTests.java

```diff
@@ -1412,6 +1413,7 @@ public void testConcurrentSegmentSearchSearchContext() throws IOException {
             .prepareUpdateSettings()
             .setTransientSettings(Settings.builder().putNull(SearchService.CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING.getKey()))
             .get();
+        assertSettingDeprecationsAndWarnings(new Setting[] { IndexSettings.INDEX_CONCURRENT_SEGMENT_SEARCH_SETTING });
```

Comment:
I added updating cluster setting of mode for existing test. 
For specific mode setting test, I have made it randomized instead of well defined scenarios since i also wanted to also check with aggregations enabled/disabled for auto mode.